src/SeqCli/Forwarder/Channel/ForwardingChannelMap.cs

```diff
@@ -45,29 +53,78 @@ ForwardingChannel OpenOrCreateChannel(string? apiKey, string name)
     }
-
-    public ForwardingChannel Get(string? apiKey)
+    
+    void LoadChannels()
     {
-        if (string.IsNullOrWhiteSpace(apiKey))
+        if (_config.Forwarder.UseApiKeyForwarding)
         {
-            return _defaultChannel;
+            foreach (var directoryPath in Directory.EnumerateDirectories(_bufferPath))
+            {
+                if (directoryPath.Equals(GetStorePath(SeqCliConnectionChannelName)))
+                {
+                    // data was stored when not using API key forwarding
+                    continue;
+                }
+
+                var path = new SystemStoreDirectory(directoryPath);
+                var apiKey = path.ReadApiKey(_config);
+
+                var channelName = ApiKeyToName(apiKey);
+                var created = OpenOrCreateChannel(apiKey, channelName);
+                _channelsByName.Add(channelName, created);
+            }
         }
-        
+        else
+        {
+            _seqCliConnectionChannel = OpenOrCreateChannel(_seqCliApiKey, SeqCliConnectionChannelName);
+        }
+    }
+
+    string GetStorePath(string name)
+    {
+        return Path.Combine(_bufferPath, name);
+    }
+
+    public ForwardingChannel GetApiKeyForwardingChannel(string? requestApiKey)
+    {
         lock (_channelsSync)
         {
-            if (_channels.TryGetValue(apiKey, out var channel))
+            var channelName = ApiKeyToName(requestApiKey);
+            
+            if (_channelsByName.TryGetValue(channelName, out var channel))
             {
                 return channel;
             }
 
-            // Seq API keys begin with four identifying characters that aren't considered part of the
-            // confidential key. TODO: we could likely do better than this.
-            var name = apiKey[..4];
-            var created = OpenOrCreateChannel(apiKey, name);
-            _channels.Add(apiKey, created);
+            var created = OpenOrCreateChannel(requestApiKey, channelName);
+            var store = new SystemStoreDirectory(GetStorePath(channelName));
+            if (requestApiKey != null)
+            {
+                store.WriteApiKey(_config, requestApiKey);
+            }
+            _channelsByName.Add(channelName, created);
             return created;
         }
     }
+    
+    public ForwardingChannel GetSeqCliConnectionChannel()
+    {
+        lock (_channelsSync)
+        {
+            if (_seqCliConnectionChannel == null)
+            {
+                _seqCliConnectionChannel = OpenOrCreateChannel(_seqCliApiKey, SeqCliConnectionChannelName);
+            }
+            return _seqCliConnectionChannel;
+        }
+    }
+
+    string ApiKeyToName(string? apiKey)
+    {
+        // Seq API keys begin with four identifying characters that aren't considered part of the
+        // confidential key. TODO: we could likely do better than this.
+        return string.IsNullOrEmpty(apiKey) ? "EmptyApiKey" : apiKey[..(Math.Min(apiKey.Length, 4))];
```

Comment:
Perhaps `EmptyApiKey` should also get a constant, and we should also disallow its use as an argument to `GetApiKeyForwardingChannel`?